### PR TITLE
Assignment 3 completed

### PR DESCRIPTION
**Pull request by ansul90**
Hi Team,

I am not sure why are my previous commits coming too. Let me know if I need to do anything regarding this.

Thanks,
Ansul

### CODE REVIEW

**Comment by vramana** [comment]
Please don't create PR's from master branch. First create a new branch and send PR from that.

**Comment by ansul90** [comment]
@vramana Where should I create a new branch?

**Comment by vramana** [comment]
Locally you should create a new branch using `git branch my-new-branch` add your work and then commit + push it and then send that branch as a PR. 

After your PR is merged, you have to switch to master branch with `git checkout master` and then do `git pull origin master`. Then your local master branch has all the new change then repeat the cycle. This is just a rough guide. You can read more about it online.

**Comment by ansul90** [comment]
ok sure, thanks @vramana

**Comment by vramana** [comment]
Remember you should never commit on master directly. 

In your case delete your forked repo and do a fresh fork again and then commit your files Assignment 3 files in the way I described. There are lots of other ways to do this but this is the easiest to explain.

**Comment by shivakrishna9** [comment]
https://github.com/shivakrishna9/learn-git/wiki/Use-a-git-branch-to-merge-a-file check this @ansul90

File: ansul/Assignment 3/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        setContentView(R.layout.activity_main);
+        Button button=(Button) findViewById(R.id.activity2_button);
+        button.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View v){
+                try{
+                    Intent intent = new Intent(MainActivity.this, SecondActivity.class);
+                    displayToast("Going to activity 2");
+                    startActivity(intent);
+                }catch (Exception e){
```
Comment (M-ZubairAhmed) [inline ansul/Assignment 3/MainActivity.java:25]: Its generally a bad idea to include whole Exception class like this also if you are planning to handle them , then be sure to do that.

File: ansul/Assignment 3/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            }
+        });
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        displayToast("Activity 1 onStart");
+    }
+
```
Comment (M-ZubairAhmed) [inline ansul/Assignment 3/MainActivity.java:37]: you could also have included onRestart to see when that event gets executed. And have you looked at other lifecycle events such as onSavedInstance, onRestoreInstance?

**Review by M-ZubairAhmed — COMMENTED** [review]
general comments - zubair

File: ansul/Assignment 3/MainActivity.java (hunk 1 above)
Comment (ansul-ng) [reply ansul/Assignment 3/MainActivity.java:25]: Right now removing the try-catch block

File: ansul/Assignment 3/MainActivity.java (hunk 2 above)
Comment (ansul-ng) [reply ansul/Assignment 3/MainActivity.java:37]: Yes, added onRestart method as well, but didn't commit that. Yes, I have looked into onSavedInstanceState() and onRestoreInstance() too. Thank you

**Pull request merged**